x-pack/solutions/observability/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -168,30 +177,26 @@ export const createLogThresholdExecutor =
             relativeViewInAppUrl
           );
 
+          const groups = getFormattedGroups(flattenGrouping);
+          const grouping = unflattenGrouping(flattenGrouping);
+
           const context = {
             ...actionContext,
             timestamp: startedAt.toISOString(),
             viewInAppUrl,
             alertDetailsUrl: getAlertDetailsUrl(libs.basePath, spaceId, uuid),
+            grouping,
           };
 
-          const instances = alertInstanceId.split(',');
-          const groups =
-            alertInstanceId !== '*'
-              ? params.groupBy?.reduce<Group[]>((resultGroups, groupByItem, index) => {
-                  resultGroups.push({ field: groupByItem, value: instances[index].trim() });
-                  return resultGroups;
-                }, [])
-              : undefined;
-
           const payload = {
             [ALERT_EVALUATION_THRESHOLD]: threshold,
             [ALERT_EVALUATION_VALUE]: value,
             [ALERT_REASON]: reason,
             [ALERT_CONTEXT]: alertContext,
             [ALERT_GROUP]: groups,
+            [ALERT_GROUPING]: grouping,
             ...flattenAdditionalContext(rootLevelContext),
-            ...getEcsGroups(groups),
+            ...getEcsGroupsFromFlattenGrouping(grouping),
```

Comment:
@cesco-f Shouldn't you pass here the `flattenGrouping`? Name of the function indicates that it expects the flatten grouping, no? Can you clarify?

Reply:
You're right, good catch, I missed that when I rebased the changes from the other PR. Changed it.